In [68]:
import numpy as np
import pandas as pd

In [69]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


**FIRSTLY IMPORT ALL MY FILES REALATED ASIA CUP**

In [70]:
df=pd.read_csv("/content/drive/MyDrive/Datasets/asiacup.csv")
bo=pd.read_csv("/content/drive/MyDrive/Datasets/batsman data odi.csv")
bt=pd.read_csv("/content/drive/MyDrive/Datasets/batsman data t20i.csv")
Bo=pd.read_csv("/content/drive/MyDrive/Datasets/bowler data odi.csv")
Bt=pd.read_csv("/content/drive/MyDrive/Datasets/bowler data t20i.csv")
wt=pd.read_csv("/content/drive/MyDrive/Datasets/wicketkeeper data odi.csv")
cp=pd.read_csv("/content/drive/MyDrive/Datasets/wicketkeeper data t20i.csv")

In [71]:
bo=pd.read_csv("/content/drive/MyDrive/Datasets/batsman data odi.csv")
bo.head()
df.head()

Team   Opponent Format   Ground  Year  Toss Selection  Run Scored  \
0   Pakistan  Sri Lanka    ODI  Sharjah  1984  Lose   Batting       187.0   
1  Sri Lanka   Pakistan    ODI  Sharjah  1984   Win   Bowling       190.0   
2      India  Sri Lanka    ODI  Sharjah  1984   Win   Bowling        97.0   
3  Sri Lanka      India    ODI  Sharjah  1984  Lose   Batting        96.0   
4      India   Pakistan    ODI  Sharjah  1984   Win   Batting       188.0   

   Wicket Lost  Fours  Sixes  Extras  Run Rate  Avg Bat Strike Rate  \
0          9.0    9.0    3.0    21.0      4.06                52.04   
1          5.0   11.0    1.0    26.0      4.36                68.51   
2          0.0    9.0    0.0    14.0      4.47                60.48   
3         10.0    7.0    0.0     8.0      2.34                25.74   
4          4.0   13.0    3.0    17.0      4.08                60.21   

   Highest Score  Wicket Taken  Given Extras  Highest Individual wicket  \
0           47.0           5.0          26.0                        2.0   
1           57.0           9.0          21.0                        3.0   
2           51.0          10.0           8.0                        3.0   
3           38.0           0.0          14.0                        0.0   
4           56.0          10.0           5.0                        3.0   

  Player Of The Match Result  
0            Roy Dias   Lose  
1            Roy Dias    Win  
2     Surinder Khanna    Win  
3     Surinder Khanna   Lose  
4     Surinder Khanna    Win

In [72]:
bo.info()
bo.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Player Name      50 non-null     object 
 1   Country          50 non-null     object 
 2   Time Period      50 non-null     object 
 3   Matches          50 non-null     int64  
 4   Played           50 non-null     int64  
 5   Not Outs         50 non-null     int64  
 6   Runs             50 non-null     int64  
 7   Highest Score    50 non-null     int64  
 8   Batting Average  50 non-null     float64
 9   Balls Faced      50 non-null     int64  
 10  Strike Rate      50 non-null     float64
 11  Centuries        50 non-null     int64  
 12  Fifties          50 non-null     int64  
 13  Ducks            50 non-null     int64  
 14  Fours            50 non-null     int64  
 15  Sixes            50 non-null     int64  
dtypes: float64(2), int64(11), object(3)
memory usage: 6.4+ KB


Matches     Played   Not Outs         Runs  Highest Score  \
count  50.000000  50.000000  50.000000    50.000000      50.000000   
mean   14.060000  13.120000   2.100000   487.340000      99.360000   
std     5.563768   5.137457   1.887121   215.852237      27.467799   
min     4.000000   3.000000   0.000000   250.000000      50.000000   
25%    10.250000   9.250000   1.000000   317.000000      76.500000   
50%    13.000000  12.000000   2.000000   443.500000     100.000000   
75%    17.750000  14.750000   3.000000   586.500000     115.500000   
max    28.000000  26.000000   6.000000  1220.000000     183.000000   

       Batting Average  Balls Faced  Strike Rate  Centuries    Fifties  \
count        50.000000    50.000000    50.000000  50.000000  50.000000   
mean         50.031000   582.200000    83.677800   0.920000   3.160000   
std          32.193085   228.148537    15.652533   1.209486   1.909803   
min          21.000000   281.000000    55.600000   0.000000   0.000000   
25%          35.482500   410.500000    76.275000   0.000000   2.000000   
50%          44.420000   519.000000    82.955000   1.000000   3.000000   
75%          58.125000   677.000000    89.792500   1.000000   4.000000   
max         251.000000  1272.000000   140.740000   6.000000   8.000000   

           Ducks       Fours      Sixes  
count  50.000000   50.000000  50.000000  
mean    0.580000   42.780000   6.660000  
std     0.702474   25.264551   5.542452  
min     0.000000    6.000000   0.000000  
25%     0.000000   26.250000   3.000000  
50%     0.000000   38.500000   5.000000  
75%     1.000000   50.500000   8.750000  
max     3.000000  139.000000  26.000000

In [73]:
bo["Batting Average"].mean()

50.031000000000006

**Batting Average By Players In Asia Cup Is 50.031000000000006 In ODI Format**


In [74]:
bo[bo["Batting Average"]== bo["Batting Average"].max()]

Player Name   Country Time Period  Matches  Played  Not Outs  Runs  \
48  Fawad Alam   Pakistan   2008-2014        4       3         2   251   

    Highest Score  Batting Average  Balls Faced  Strike Rate  Centuries  \
48            114            251.0          281        89.32          1   

    Fifties  Ducks  Fours  Sixes  
48        2      0     15      5

**FAWAD ALAM HAVE BEST BATTING AVERAGE IN ASIA CUP**

In [75]:
bo[bo["Strike Rate"]== bo["Strike Rate"].max()]

Player Name   Country Time Period  Matches  Played  Not Outs  Runs  \
17  Shahid Afridi   Pakistan   1997-2014       23      21         6   532   

    Highest Score  Batting Average  Balls Faced  Strike Rate  Centuries  \
17            124            35.46          378       140.74          2   

    Fifties  Ducks  Fours  Sixes  
17        1      2     46     26

**SHAHID AFRIDI HAS BEST STRIKE RATE IN ASIA CUP**

In [76]:
bo[bo["Runs"]==bo["Runs"].max()]

Player Name    Country Time Period  Matches  Played  Not Outs  Runs  \
0  ST Jayasuriya   Sri Lanka   1990-2008       25      24         1  1220   

   Highest Score  Batting Average  Balls Faced  Strike Rate  Centuries  \
0            130            53.04         1190       102.52          6   

   Fifties  Ducks  Fours  Sixes  
0        3      1    139     23

**ST Jayasuriya has most runs in ASIA CUP**

In [77]:
cp=pd.read_csv("/content/drive/MyDrive/Datasets/champion.csv")
cp.head()

Year        Host  No Of Team   Champion  Runner Up Player Of The Series  \
0  1984         UAE           3      India  Sri Lanka      Surinder Khanna   
1  1986   Sri Lanka           3  Sri Lanka   Pakistan     Arjuna Ranatunga   
2  1988  Bangladesh           4      India  Sri Lanka         Navjot Sidhu   
3  1990       India           3      India  Sri Lanka          Not Awarded   
4  1995         UAE           4      India  Sri Lanka         Navjot Sidhu   

  Highest Run Scorer Highest Wicket Taker  
0    Surinder Khanna         Ravi Shastri  
1   Arjuna Ranatunga          Abdul Qadir  
2         Ijaz Ahmed          Arshad Ayub  
3   Arjuna Ranatunga            Kapil Dev  
4   Sachin Tendulkar          Anil Kumble

In [78]:
Bo.head()

Player Name    Country Time Period  Matches  Played  Overs  \
0  M Muralidaran   Sri Lanka   1995-2010       24      24  230.2   
1     SL Malinga   Sri Lanka   2004-2018       14      14  128.1   
2     BAW Mendis   Sri Lanka   2008-2014        8       8   68.0   
3    Saeed Ajmal    Pakistan   2008-2014       12      12  115.0   
4     WPUJC Vaas   Sri Lanka   1995-2008       19      19  152.2   

   Maiden Overs  Runs  Wickets Best Figure  Bowling Average  Economy Rate  \
0            13   865       30        5/31            28.83          3.75   
1             6   596       29        5/34            20.55          4.65   
2             5   271       26        6/13            10.42          3.98   
3             6   485       25        3/26            19.40          4.21   
4            20   639       23        3/30            27.78          4.19   

   Strike Rate  Four Wickets  Five Wickets  
0         46.0             1             1  
1         26.5             1             3  
2         15.6             2             2  
3         27.6             0             0  
4         39.7             0             0

In [79]:
Bo[Bo["Wickets"]==Bo["Wickets"].max()]

Player Name    Country Time Period  Matches  Played  Overs  \
0  M Muralidaran   Sri Lanka   1995-2010       24      24  230.2   

   Maiden Overs  Runs  Wickets Best Figure  Bowling Average  Economy Rate  \
0            13   865       30        5/31            28.83          3.75   

   Strike Rate  Four Wickets  Five Wickets  
0         46.0             1             1

**M Muralidaran was  most wicket taker bowler in ODI Format**

In [80]:
Bo[Bo["Economy Rate"]==Bo['Economy Rate'].min()]

Player Name   Country Time Period  Matches  Played  Overs  Maiden Overs  \
13  Wasim Akram   Pakistan   1986-2000       12      12   92.1             8   

    Runs  Wickets Best Figure  Bowling Average  Economy Rate  Strike Rate  \
13   297       16        4/19            18.56          3.22         34.5   

    Four Wickets  Five Wickets  
13             1             0

****

**Wasim Akram was the most economical bowler is ODI format**

In [81]:
Bo[Bo["Bowling Average"]==Bo['Bowling Average'].min()]

Player Name    Country Time Period  Matches  Played  Overs  Maiden Overs  \
2  BAW Mendis   Sri Lanka   2008-2014        8       8   68.0             5   

   Runs  Wickets Best Figure  Bowling Average  Economy Rate  Strike Rate  \
2   271       26        6/13            10.42          3.98         15.6   

   Four Wickets  Five Wickets  
2             2             2

**BAW Mendis was the avarage  bowler**

In [82]:
Bo[Bo["Maiden Overs"]==Bo['Maiden Overs'].max()]

Player Name    Country Time Period  Matches  Played  Overs  Maiden Overs  \
4  WPUJC Vaas   Sri Lanka   1995-2008       19      19  152.2            20   

   Runs  Wickets Best Figure  Bowling Average  Economy Rate  Strike Rate  \
4   639       23        3/30            27.78          4.19         39.7   

   Four Wickets  Five Wickets  
4             0             0

**WPUJC Vaas bowled most maiden overs in ODI format**

In [83]:
Bo[Bo["Runs"]==Bo['Runs'].min()]

Player Name Country Time Period  Matches  Played  Overs  Maiden Overs  \
37  Arshad Ayub    India   1988-1988        4       4   36.0             2   

    Runs  Wickets Best Figure  Bowling Average  Economy Rate  Strike Rate  \
37   120        9        5/21            13.33          3.33         24.0   

    Four Wickets  Five Wickets  
37             0             1

**Arshad Ayub considered less runs in ODI format**

In [84]:
bt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49 entries, 0 to 48
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Player Name      49 non-null     object 
 1   Country          49 non-null     object 
 2   Time Period      49 non-null     object 
 3   Matches          49 non-null     int64  
 4   Played           49 non-null     int64  
 5   Not Outs         49 non-null     int64  
 6   Runs             49 non-null     int64  
 7   Highest Score    49 non-null     int64  
 8   Batting Average  49 non-null     object 
 9   Balls Faced      49 non-null     int64  
 10  Strike Rate      49 non-null     float64
 11  Centuries        49 non-null     int64  
 12  Fifties          49 non-null     int64  
 13  Ducks            49 non-null     int64  
 14  Fours            49 non-null     int64  
 15  Sixes            49 non-null     int64  
dtypes: float64(1), int64(11), object(4)
memory usage: 6.2+ KB


In [85]:
bt.head()

Player Name      Country Time Period  Matches  Played  Not Outs  Runs  \
0          V Kohli         India   2016-2022       10       9         4   429   
1  Mohammad Rizwan      Pakistan   2022-2022        6       6         1   281   
2        RG Sharma         India   2016-2022        9       9         0   271   
3      Babar Hayat     Hong Kong   2016-2022        5       5         0   235   
4   Ibrahim Zadran   Afghanistan   2022-2022        5       5         2   196   

   Highest Score Batting Average  Balls Faced  Strike Rate  Centuries  \
0            122           85.80          325       132.00          1   
1             78           56.20          239       117.57          0   
2             83           30.11          192       141.14          0   
3            122           47.00          160       146.87          1   
4             64           65.33          188       104.25          0   

   Fifties  Ducks  Fours  Sixes  
0        3      1     40     11  
1        3      0     21      6  
2        2      1     27     12  
3        1      1     22     10  
4        1      0     14      4

In [86]:
bt[bt['Batting Average']==bt['Batting Average'].max()]

Player Name Country Time Period  Matches  Played  Not Outs  Runs  \
0    V Kohli    India   2016-2022       10       9         4   429   

   Highest Score Batting Average  Balls Faced  Strike Rate  Centuries  \
0            122           85.80          325        132.0          1   

   Fifties  Ducks  Fours  Sixes  
0        3      1     40     11

**Virat kohli have best batting average in t20 format of asia cup**

In [87]:
bt[bt['Strike Rate']==bt['Strike Rate'].max()]

Player Name     Country Time Period  Matches  Played  Not Outs  \
38  Mosaddek Hossain   Bangladesh   2022-2022        2       2         2   

    Runs  Highest Score Batting Average  Balls Faced  Strike Rate  Centuries  \
38    72             48               -           40        180.0          0   

    Fifties  Ducks  Fours  Sixes  
38        0      0      8      1

**Mosaddek Hossain has best strike rate in t20 format of asia cup**

In [88]:
bt[bt['Runs']==bt['Runs'].max()]

Player Name Country Time Period  Matches  Played  Not Outs  Runs  \
0    V Kohli    India   2016-2022       10       9         4   429   

   Highest Score Batting Average  Balls Faced  Strike Rate  Centuries  \
0            122           85.80          325        132.0          1   

   Fifties  Ducks  Fours  Sixes  
0        3      1     40     11

**Virat kohali have most runs in t20 format of asia cup**

In [89]:
bt[bt['Centuries']==bt['Centuries'].max()]

Player Name    Country Time Period  Matches  Played  Not Outs  Runs  \
0      V Kohli       India   2016-2022       10       9         4   429   
3  Babar Hayat   Hong Kong   2016-2022        5       5         0   235   

   Highest Score Batting Average  Balls Faced  Strike Rate  Centuries  \
0            122           85.80          325       132.00          1   
3            122           47.00          160       146.87          1   

   Fifties  Ducks  Fours  Sixes  
0        3      1     40     11  
3        1      1     22     10

**Virat Kohli And Babar Hayat have most centuries in t20 format of asia cup (1 century each)**

****

In [90]:
cp.Champion.value_counts()

India        7
Sri Lanka    6
Pakistan     2
Name: Champion, dtype: int64

**INDIA IS MOST TIME CHAMPION IN ASIA CUP**

In [91]:
wo=pd.read_csv("/content/drive/MyDrive/Datasets/wicketkeeper data odi.csv")
wo.head()

Player Name     Country Time Period  Matches  Played  Dismissals  \
0         MS Dhoni        India   2008-2018       19      19          36   
1    KC Sangakkara    Sri Lanka   2004-2014       24      24          36   
2        Moin Khan     Pakistan   1995-2004       14      13          17   
3  Mushfiqur Rahim   Bangladesh   2008-2018       21      17          17   
4     DSBP Kuruppu    Sri Lanka   1984-1988        9       9          14   

   Catches  Stumpings  Maximum Dismissals  
0       25         11                   5  
1       27          9                   4  
2       12          5                   3  
3       14          3                   4  
4       12          2                   4

In [92]:
wo[wo["Dismissals"]==wo["Dismissals"].max()]

Player Name    Country Time Period  Matches  Played  Dismissals  \
0       MS Dhoni       India   2008-2018       19      19          36   
1  KC Sangakkara   Sri Lanka   2004-2014       24      24          36   

   Catches  Stumpings  Maximum Dismissals  
0       25         11                   5  
1       27          9                   4

**WITH OUT ANY DOUBT MSD IS THE BEST WICKET KEEPER**